--- zlib-1.2.8/win32/Makefile.gcc.orig	2016-02-11 13:46:54.000000000 -0500
+++ zlib-1.2.8/win32/Makefile.gcc	2016-02-11 13:53:08.000000000 -0500
@@ -57,6 +57,8 @@
 RC = $(PREFIX)windres
 RCFLAGS = --define GCC_WINDRES

+IMAGE_BASE = 0x63300000
+
 STRIP = $(PREFIX)strip

 CP = cp -fp
@@ -93,7 +95,7 @@
 $(IMPLIB): $(SHAREDLIB)

 $(SHAREDLIB): win32/zlib.def $(OBJS) $(OBJA) zlibrc.o
-	$(CC) -shared -Wl,--out-implib,$(IMPLIB) $(LDFLAGS) \
+	$(CC) -shared -Wl,--out-implib,$(IMPLIB),--image-base,$(IMAGE_BASE) $(LDFLAGS) \
 	-o $@ win32/zlib.def $(OBJS) $(OBJA) zlibrc.o
 	$(STRIP) $@
