--- ruby-2.1.2/vm_core.h.orig	2014-09-29 14:05:24.000000000 -0500
+++ ruby-2.1.2/vm_core.h	2014-09-29 14:05:39.000000000 -0500
@@ -392,9 +392,9 @@
 
     /* postponed_job */
     struct rb_postponed_job_struct *postponed_job_buffer;
-    int postponed_job_index;
+    long postponed_job_index;
 
-    int src_encoding_index;
+    long src_encoding_index;
 
     VALUE verbose, debug, orig_progname, progname;
     VALUE coverages;
