--- ruby-1.9.3-p547/configure.orig	2014-05-16 09:38:31 -0500
+++ ruby-1.9.3-p547/configure	2014-07-15 19:58:29 -0500
@@ -16488,6 +16488,7 @@
   aix*) :
     	: ${LDSHARED='$(CC)'}
 			LDSHARED="$LDSHARED ${linker_flag}-G"
+			DLDFLAGS='-eInit_$(TARGET)'
 			EXTDLDFLAGS='-e$(TARGET_ENTRY)'
 			XLDFLAGS="${linker_flag}"'-bE:$(ARCHFILE)'" ${linker_flag}-brtl"
 			XLDFLAGS="$XLDFLAGS ${linker_flag}-blibpath:${prefix}/lib:${LIBPATH:-/usr/lib:/lib}"
@@ -17028,7 +17029,8 @@
 
 	LIBRUBY_DLDFLAGS="${linker_flag}-bnoentry $XLDFLAGS"
 	LIBRUBYARG_SHARED='-L${libdir} -l${RUBY_SO_NAME}'
-	SOLIBS='-lm -lc'
+	SOLIBS='-lm -lc -lz'
+        LIBRUBY_SO='lib$(RUBY_SO_NAME).a'
 	 ;; #(
   beos*) :
 
