--- bzip2-1.0.8/Makefile-libbz2_so.orig	2019-07-24 10:33:36.184131389 -0400
+++ bzip2-1.0.8/Makefile-libbz2_so	2019-07-24 10:35:20.564277676 -0400
@@ -35,7 +35,7 @@
       bzlib.o
 
 all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.8 $(OBJS)
+	$(CC) -qmkshrobj -Wl,-soname -o libbz2.so.1.0.8 $(OBJS)
 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.8
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.8 libbz2.so.1.0
