diff --git a/Modules/Setup b/Modules/Setup
index 02cfb67518..634467ef20 100644
--- a/Modules/Setup
+++ b/Modules/Setup
@@ -366,7 +366,8 @@ xxsubtype xxsubtype.c
 # Uncommenting the following line tells makesetup that all following modules
 # are not built (see above for more detail).
 #
-#*disabled*
+*disabled*
+nis nismodule.c
 #
 #_sqlite3 _tkinter _curses pyexpat
 #_codecs_jp _codecs_kr _codecs_tw unicodedata
