--- openssl-1.1.0f/Configurations/unix-Makefile.tmpl.orig	2017-10-22 13:23:48.592533617 +0300
+++ openssl-1.1.0f/Configurations/unix-Makefile.tmpl	2017-10-22 13:26:21.199065333 +0300
@@ -260,7 +260,7 @@
 	@echo "Tests are not supported with your chosen Configure options"
 	@ : {- output_on() if !$disabled{tests}; "" -}

-install: install_sw install_ssldirs install_docs
+install: install_sw install_ssldirs

 uninstall: uninstall_docs uninstall_sw
